textacy/corpora/wiki_reader.py

```diff
@@ -0,0 +1,212 @@
+                        section_idx += 1
+                    except IndexError:
+                        continue
+
+        return parsed_page
+
+    def texts(self, min_len=100, limit=-1):
+        """
+        Iterate over the pages in a Wikipedia articles database dump (*articles.xml.bz2),
+        yielding the plain text of a page, one at a time.
+
+        Args:
+            min_len (int): minimum length in chars that a page must have
+                for it to be returned; too-short pages are skipped (optional)
+            limit (int): maximum number of pages (passing `min_len`) to yield;
+                if -1, all pages in the db dump are iterated over (optional)
+
+        Yields:
+            str: plain text for the next page in the wikipedia database dump
+
+        Notes:
+            .. Page and section titles appear immediately before the text content
+               that they label, separated by a single newline character.
+        """
+        n_pages = 0
+        for _, title, content in self:
+            text = self._clean_content(content)
+            if len(text) < min_len:
+                continue
+
+            yield title + '\n' + text
+
+            n_pages += 1
+            if n_pages == limit:
+                break
+
+    def pages(self, min_len=100, limit=-1, metadata=True):
+        """
+        Iterate over the pages in a Wikipedia articles database dump (*articles.xml.bz2),
+        yielding one page whose structure and content have been parsed, as a dict.
+
+        Args:
+            min_len (int): minimum length in chars that a page must have
+                for it to be returned; too-short pages are skipped
+            limit (int): maximum number of pages (passing ``min_len``) to yield;
+                if -1, all pages in the db dump are iterated over (optional)
+            metadata (bool): if True, return page metadata in addition to content,
+                including 'ext_links', 'wiki_links', and 'categories' (optional)
+
+        Yields:
+            dict: the next page's parsed content, including 'title' and 'page_id'
+
+                Key 'sections' includes a list of all page sections, each with 'title'
+                for the section title, 'text' for plain text content,'idx' for position
+                on page, and 'level' for the depth of the section within the page's hierarchy
+
+                If ``metadata`` is True, there are additional keys: 'wiki_links' is
+                a list of _other_ page titles linked to from this page; 'ext_links' is
+                a list of external URLs linked to from this page; and 'categories' is
+                a list of Wikipedia categories to which this page belongs.
+
+        Notes:
+            .. This function requires `mwparserfromhell <mwparserfromhell.readthedocs.org>`_
+        """
+        import mwparserfromhell  # hiding this here; don't want another required dep
```

Comment:
No; I should add a try/except ImportError block, with a helpful error message. Will add.